pkg/kv/kvserver/store_create_replica.go

```diff
@@ -361,17 +361,17 @@ func fromReplicaIsTooOldRLocked(toReplica *Replica, fromReplica *roachpb.Replica
 	return !found && fromReplica.ReplicaID < desc.NextReplicaID
 }
 
-// addReplicaInternalLocked adds the replica to the replicas map and the
-// replicasByKey btree. Returns an error if a replica with
-// the same Range ID or an overlapping replica or placeholder exists in
-// this store. addReplicaInternalLocked requires that the store lock is held.
+// addReplicaInternalLocked adds the replica to the replicasByKey btree. The
+// replica must already be in replicasByRangeID. Requires that Store.mu is held.
+//
+// Returns an error if a different replica with the same range ID, or an
+// overlapping replica or placeholder exists in this Store.
 func (s *Store) addReplicaInternalLocked(repl *Replica) error {
```

Comment:
It's because of https://github.com/cockroachdb/cockroach/issues/94912, the insertion happens in that method: https://github.com/cockroachdb/cockroach/blob/3d51c79d527871333bb690220da2d0ac90fc793d/pkg/kv/kvserver/store_create_replica.go#L448-L453

Reply:
Updated the commit message.